Bottom: fe0c91d69c7a2182d2f1441d54f1c31813baf5fe
Top:    3f6c28485db042d2b6e315b33805f1d7f15043ed
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-03-12 01:30:59 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 9760ab0..870908d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1154,7 +1154,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
